app/src/main/java/org/mozilla/fenix/customtabs/FennecWebAppIntentProcessor.kt

```diff
@@ -102,10 +102,17 @@ class FennecWebAppIntentProcessor(
     internal fun fromFile(path: String?): WebAppManifest? {
         if (path.isNullOrEmpty()) return null
 
+        // Fennec manifests should be located in <filesDir>/mozilla/<profile>/manifests/
+        val file = File(path)
+        val manifestsDir = file.parentFile
+        // Check that manifest is in a folder named "manifests"
+        if (manifestsDir == null || manifestsDir.name != "manifests" ||
+            manifestsDir.parentFile?.parentFile != getMozillaDirectory()) return null
```

Comment:
Maybe we should compare the canonical path (`getCanonicalPath()/getCanonicalFile()`) here, to make sure we match on the resolved paths. Afaik @Amejia481 recently wrote path comparison code too and may have input here.

Reply:
Yes, agree we did something similar on [isUnderPrivateAppDirectory](https://github.com/mozilla-mobile/android-components/blob/1d5e94f0d2dac22f9b356280e0be6256270d93f7/components/support/ktx/src/main/java/mozilla/components/support/ktx/android/net/Uri.kt#L65)